### ISSUE TITLE
_convert_to_request_dict fix introduced error in urllib

### ISSUE THREAD

**Issue by notenti** [pynamodb/PynamoDB#1085]
I recently installed 5.2.2 to test the fix implemented in https://github.com/pynamodb/PynamoDB/pull/1083. Unfortunately, it looks like there's a new issue.

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "venv/lib/python3.9/site-packages/pynamodb/pagination.py", line 194, in __next__
    self._get_next_page()
  File "venv/lib/python3.9/site-packages/pynamodb/pagination.py", line 179, in _get_next_page
    page = next(self.page_iter)
  File "venv/lib/python3.9/site-packages/pynamodb/pagination.py", line 115, in __next__
    page = self._operation(*self._args, settings=self._settings, **self._kwargs)
  File "venv/lib/python3.9/site-packages/pynamodb/connection/table.py", line 238, in scan
    return self.connection.scan(
  File "venv/lib/python3.9/site-packages/pynamodb/connection/base.py", line 1265, in scan
    return self.dispatch(SCAN, operation_kwargs, settings)
  File "venv/lib/python3.9/site-packages/pynamodb/connection/base.py", line 332, in dispatch
    data = self._make_api_call(operation_name, operation_kwargs, settings)
  File "venv/lib/python3.9/site-packages/pynamodb/connection/base.py", line 361, in _make_api_call
    request_dict = self.client._convert_to_request_dict(
  File "venv/lib/python3.9/site-packages/botocore/client.py", line 984, in _convert_to_request_dict
    prepare_request_dict(
  File "venv/lib/python3.9/site-packages/botocore/awsrequest.py", line 270, in prepare_request_dict
    url = _urljoin(endpoint_url, r['url_path'], host_prefix)
  File "venv/lib/python3.9/site-packages/botocore/awsrequest.py", line 336, in _urljoin
    reconstructed = urlunsplit((p[0], new_netloc, new_path, p[3], p[4]))
  File ".pyenv/versions/3.9.7/lib/python3.9/urllib/parse.py", line 512, in urlunsplit
    _coerce_args(*components))
  File ".pyenv/versions/3.9.7/lib/python3.9/urllib/parse.py", line 125, in _coerce_args
    raise TypeError("Cannot mix str and non-str arguments")
TypeError: Cannot mix str and non-str arguments
```

**Comment by guymowbray** [comment]
Same we are getting this message: 
```
model = next(query)  File "/var/task/pynamodb/pagination.py", line 194, in __next__    self._get_next_page()  File "/var/task/pynamodb/pagination.py", line 179, in _get_next_page    page = next(self.page_iter)  File "/var/task/pynamodb/pagination.py", line 115, in __next__    page = self._operation(*self._args, settings=self._settings, **self._kwargs)  File "/var/task/pynamodb/connection/table.py", line 270, in query    return self.connection.query(  File "/var/task/pynamodb/connection/base.py", line 1295, in query    tbl = self.get_meta_table(table_name)  File "/var/task/pynamodb/connection/base.py", line 555, in get_meta_table    data = self.dispatch(DESCRIBE_TABLE, operation_kwargs)  File "/var/task/pynamodb/connection/base.py", line 332, in dispatch    data = self._make_api_call(operation_name, operation_kwargs, settings)  File "/var/task/pynamodb/connection/base.py", line 361, in _make_api_call    request_dict = self.client._convert_to_request_dict(  File "/var/task/botocore/client.py", line 984, in _convert_to_request_dict    prepare_request_dict(  File "/var/task/botocore/awsrequest.py", line 270, in prepare_request_dict    url = _urljoin(endpoint_url, r['url_path'], host_prefix)  File "/var/task/botocore/awsrequest.py", line 336, in _urljoin    reconstructed = urlunsplit((p[0], new_netloc, new_path, p[3], p[4]))  File "/var/lang/lib/python3.8/urllib/parse.py", line 502, in urlunsplit    _coerce_args(*components))  File "/var/lang/lib/python3.8/urllib/parse.py", line 124, in _coerce_args    raise TypeError("Cannot mix str and non-str arguments") | [ERROR] TypeError: Cannot mix str and non-str arguments Traceback (most recent call last):   File "/var/task/forecasttooling/aws_lambda.py", line 18, in wrapper     raise error   File "/var/task/forecasttooling/aws_lambda.py", line 15, in wrapper     return func(event, _context)   File "/var/task/salesoperationsplanning/application/salesoperationsplanningpublisher/app.py", line 184, in lambda_handler     user_inputs = get_user_inputs(menu_week)   File "/var/task/salesoperationsplanning/achievable_demand_plan/user_inputs.py", line 21, in get_user_inputs     model = next(query)   File "/var/task/pynamodb/pagination.py", line 194, in __next__     self._get_next_page()   File "/var/task/pynamodb/pagination.py", line 179, in _get_next_page     page = next(self.page_iter)   File "/var/task/pynamodb/pagination.py", line 115, in __next__     page = self._operation(*self._args, settings=self._settings, **self._kwargs)   File "/var/task/pynamodb/connection/table.py", line 270, in query     return self.connection.query(   File "/var/task/pynamodb/connection/base.py", line 1295, in query     tbl = self.get_meta_table(table_name)   File "/var/task/pynamodb/connection/base.py", line 555, in get_meta_table     data = self.dispatch(DESCRIBE_TABLE, operation_kwargs)   File "/var/task/pynamodb/connection/base.py", line 332, in dispatch     data = self._make_api_call(operation_name, operation_kwargs, settings)   File "/var/task/pynamodb/connection/base.py", line 361, in _make_api_call     request_dict = self.client._convert_to_request_dict(   File "/var/task/botocore/client.py", line 984, in _convert_to_request_dict     prepare_request_dict(   File "/var/task/botocore/awsrequest.py", line 270, in prepare_request_dict     url = _urljoin(endpoint_url, r['url_path'], host_prefix)   File "/var/task/botocore/awsrequest.py", line 336, in _urljoin     reconstructed = urlunsplit((p[0], new_netloc, new_path, p[3], p[4]))   File "/var/lang/lib/python3.8/urllib/parse.py", line 502, in urlunsplit     _coerce_args(*components))   File "/var/lang/lib/python3.8/urllib/parse.py", line 124, in _coerce_args     raise TypeError("Cannot mix str and non-str arguments")
```

**Comment by ikonst** [comment]
Oh. It looks like the fix was not so simple after all, since `endpoint_url` cannot be `None`.

**Comment by ikonst** [comment]
This should be fixed in 5.2.3.
